pkg/trtl/honu.go

```diff
@@ -101,7 +101,69 @@ func (h *HonuService) Get(ctx context.Context, in *pb.GetRequest) (*pb.GetReply,
 
 // Put is a unary request to store a value for a key.
 func (h *HonuService) Put(ctx context.Context, in *pb.PutRequest) (out *pb.PutReply, err error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	if _, found := reservedNamespaces[in.Namespace]; found {
+		log.Warn().Msg("cannot use reserved namespace")
+		return nil, status.Error(codes.PermissionDenied, "cannot use reserved namespace")
+	}
+	if len(in.Key) == 0 {
+		log.Warn().Msg("missing key in Trtl Put request")
+		return nil, status.Error(codes.InvalidArgument, "key must be provided in Put request")
+	}
+	if len(in.Value) == 0 {
+		log.Warn().Msg("missing value in Trtl Put request")
+		return nil, status.Error(codes.InvalidArgument, "value must be provided in Put request")
+	}
+
+	var key []byte
+	if len(in.Namespace) > 0 {
+		key = prepend(in.Namespace, in.Key)
+	} else {
+		key = prepend("default", in.Key)
+	}
+
+	if in.Options != nil {
+		log.Debug().Str("key", string(key)).Bool("return_meta", in.Options.ReturnMeta).Msg("Trtl Put")
+	} else {
+		log.Debug().Str("key", string(key)).Msg("Trtl Put")
+	}
+
+	var success bool
+	if err := h.db.Put(key, in.Value); err != nil {
+		success = false
+	} else {
+		success = true
+	}
+
+	out = &pb.PutReply{Success: success}
+
+	// if Options include a request for metadata, we need to do a Get
```

Comment:
Not really sure if this is what we imagined when we said we would enable Options for `Put`. Let me know if I guessed incorrectly!